io/io.go

```diff
@@ -258,6 +258,11 @@ func inferFileIOFromSchema(ctx context.Context, path string, props map[string]st
 		bucket = memblob.OpenBucket(nil)
 	case "file", "":
 		return LocalFS{}, nil
+	case "abfs", "abfss", "wasb", "wasbs":
```

Comment:
do the different schemes mean anything important?

Reply:
hmm, just try to follow the definition in iceberg-jave:
```
  private static final Map<String, String> SCHEME_TO_FILE_IO =
      ImmutableMap.of(
          "s3", S3_FILE_IO_IMPL,
          "s3a", S3_FILE_IO_IMPL,
          "s3n", S3_FILE_IO_IMPL,
          "gs", GCS_FILE_IO_IMPL,
          "abfs", ADLS_FILE_IO_IMPL,
          "abfss", ADLS_FILE_IO_IMPL,
          "wasb", ADLS_FILE_IO_IMPL,
          "wasbs", ADLS_FILE_IO_IMPL);
```

Reply:
> do the different schemes mean anything important?

The abfs URIs will use a different HTTP endpoint than wasb but they are addressing the same object.